Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -333,111 +326,229 @@ ::IsDisplayableInThreeDView(const char* viewNodeID)const
 }
 
 //----------------------------------------------------------------------------
-void vtkMRMLSliceNode::SetOrientation(const char* orientation)
+bool vtkMRMLSliceNode::SetOrientation(const char* orientation)
 {
   if (!orientation)
     {
-    return;
+    return false;
     }
- if (!strcmp(orientation, "Axial"))
+
+  vtkMatrix4x4 *orientationPreset;
+  orientationPreset = this->GetSliceOrientationPreset(orientation);
+
+  if(!strcmp(orientation, "Reformat"))
     {
-    this->SetOrientationToAxial();
+    this->SetOrientationString(orientation);
     }
-  else if (!strcmp(orientation, "Sagittal"))
+
+  if(orientationPreset == NULL)
+    {
+    return false;
+    }
+
+  //SetOrientationString() and SetOrientationReference()
+  //behaviour will be mantained for backward compatibility
```

Comment:
Do you mean that the method could be completely removed from the class ? Or do you mean calling these method below is only done to maintain backward compatibility ?